internal/manifest/docker_schema2_list.go

```diff
@@ -90,6 +99,50 @@ func (list *Schema2ListPublic) UpdateInstances(updates []ListUpdate) error {
 	return nil
 }
 
+func (index *Schema2List) EditInstances(editInstances []ListEdit) error {
+	addedEntries := []Schema2ManifestDescriptor{}
+	for i, editInstance := range editInstances {
+		switch editInstance.ListOperation {
+		case ListOpUpdate:
+			if err := editInstance.UpdateOldDigest.Validate(); err != nil {
+				return fmt.Errorf("Schema2List.EditInstances: Attempting to update %s which is an invalid digest: %w", editInstance.UpdateOldDigest, err)
+			}
+			if err := editInstance.UpdateDigest.Validate(); err != nil {
+				return fmt.Errorf("Schema2List.EditInstances: Modified digest %s is an invalid digest: %w", editInstance.UpdateDigest, err)
+			}
+			instanceToBeModified := index.Descriptor(editInstance.UpdateOldDigest)
+			if instanceToBeModified.Digest != "" {
+				return fmt.Errorf("Schema2List.EditInstances: Attempting to update %s which is an invalid digest", editInstance.UpdateOldDigest)
+			}
+			instanceToBeModified.Digest = editInstance.UpdateDigest
+			if editInstance.UpdateSize < 0 {
+				return fmt.Errorf("update %d of %d passed to Schema2List.UpdateInstances had an invalid size (%d)", i+1, len(editInstances), editInstance.UpdateSize)
+			}
+			instanceToBeModified.Size = editInstance.UpdateSize
+			if editInstance.UpdateMediaType == "" {
+				return fmt.Errorf("update %d of %d passed to Schema2List.UpdateInstances had no media type (was %q)", i+1, len(editInstances), index.Manifests[i].MediaType)
+			}
+			instanceToBeModified.MediaType = editInstance.UpdateMediaType
```

Comment:
How does editing the local `…ToBeModified` variable make any difference?

Reply:
I meant to pass by reference.